### ISSUE TITLE
Open up for other validators

### ISSUE THREAD

**Issue by JornWildt** [digst/OIOSAML.Net#23]
The OIOSAML library is actually quite a versatile tool, since SAML2 is used by other IdPs than just NemLogin. We have used it with success for authenticating with ADFS - but we were forced to use a slightly different validator.

As can be seen in https://github.com/cBrain-dk/OIOSAML.Net/commit/90d42baf359be98415189e8b92faf3463f1c88d4, we added a new attribute to FederationConfig.cs named "AssertionValidatorType". This in turn is used to instantiate other assertion validators than the built-in versions. See the change to Saml20Assertion.cs in https://github.com/cBrain-dk/OIOSAML.Net/commit/90d42baf359be98415189e8b92faf3463f1c88d4.

Further more we also reuse the existing validators by opening them up with public and protected virtual properties that allows us to create our own modified inherited versions. In our case, all we needed was to allow attributes without namespace.

I'm not creating a pull request as our changes are too unpolished to include - just supplying them as an example of what we have done.

So, in short, the issue is a wish for a more open validator structure. The current is fine, it just needs to be made public.
